File: services/platform/api/src/main/java/com/joshsoll/telemetry/platform/organizationmembership/dto/OrganizationMembershipResponse.java
package com.joshsoll.telemetry.platform.organizationmembership.dto;

import java.time.Instant;
import java.util.UUID;

import com.joshsoll.telemetry.platform.organizationmembership.enums.MembershipStatus;
import com.joshsoll.telemetry.platform.organizationmembership.enums.OrganizationRole;

public record OrganizationMembershipResponse(
                UUID id,
                UUID organizationId,
                UUID userId,

                String firstName,
                String lastName,
                String email,

                OrganizationRole role,
                MembershipStatus status,

                Instant createdAt,
                Instant updatedAt) {
}

File: services/platform/web/src/app/features/organizationMembership/dto/organization-membership-response.ts
import { OrganizationRole } from "../enums/organization-role";
import { MembershipStatus } from "../enums/membership-status";

export interface OrganizationMembershipResponse {
    id: string;
    organizationId: string;
    userId: string;

    firstName: string,
    lastName: string,
    email: string,

    role: OrganizationRole;
    status: MembershipStatus;

    createdAt: string;
    updatedAt: string;
}
